### PR TITLE
ZER-108 - Fixed issues with edit accounts

### PR DESCRIPTION
**Pull request by AnastasiiaBilynska**
dev
## JIRA

* [ZER-108](https://jira.softserve.academy/browse/ZER-108)

## Code reviewers

- [ ] @github_username

### Second Level Review

- [ ] @github_username

## Summary of issue

ToDo

## Summary of change

ToDo

## Testing approach

ToDo

## CHECK LIST
- [ ]  СI passed
- [ ]  Сode coverage >=95%
- [ ]  PR is reviewed manually again (to make sure you have 100% ready code)
- [ ]  All reviewers agreed to merge the PR
- [ ]  I've checked new feature as logged in and logged out user if needed
- [ ]  PR meets all conventions

### CODE REVIEW

File: app/controllers/admins/users_controller.rb [hunk 1]
```diff
@@ -19,7 +19,16 @@ def index
     end
 
     def update
-      if user.update(user_params)
+      updated = if user_params[:password].blank? &&
```
Comment (olegsobchuk) [inline app/controllers/admins/users_controller.rb:22]: why we can't check only `password` ?

File: app/controllers/admins/users_controller.rb [hunk 2]
```diff
@@ -19,7 +19,16 @@ def index
...
+                   user_params[:password_confirmation].blank?
+                  user_params.delete(:password)
```
Comment (olegsobchuk) [inline app/controllers/admins/users_controller.rb:24]: why do we need to delete  this?

**Review by olegsobchuk — COMMENTED** [review]
review 1

File: app/controllers/admins/users_controller.rb (hunk 2 above)
Comment (AnastasiiaBilynska) [reply app/controllers/admins/users_controller.rb:24]: We can remove that. Fixed

File: app/controllers/admins/users_controller.rb (hunk 1 above)
Comment (AnastasiiaBilynska) [reply app/controllers/admins/users_controller.rb:22]: If we check only passwor, password confirmation will be skipped

File: app/controllers/admins/users_controller.rb [hunk 3]
```diff
@@ -29,7 +29,16 @@ def update
...
-      params.require(:user).permit(:first_name, :last_name, :country, :password,:password_confirmation, :blocked, :avatar).merge(skip_password: true)
+      prms = params.require(:user).permit(:first_name,
+                                          :last_name,
+                                          :country,
+                                          :password,
+                                          :password_confirmation,
+                                          :blocked,
+                                          :avatar)
+      prms = prms.merge(skip_password_validation: true) unless prms[:password].present? ||
+      prms[:password_confirmation].present?
```
Comment (olegsobchuk) [inline app/controllers/admins/users_controller.rb:40]: remove `password_confirmation` and do this in model

**Review by olegsobchuk — APPROVED** [review]

**Review by aleksasha-o — APPROVED** [review]

**Pull request merged**